cohere/client.py

```diff
@@ -499,25 +500,85 @@ def create_cluster_job(
         threshold: Optional[float] = None,
         min_cluster_size: Optional[int] = None,
     ) -> CreateClusterJobResponse:
+        """Create clustering job.
+
+        Args:
+            embeddings_url (str): File with embeddings to cluster.
+            threshold (Optional[float], optional): Similarity threshold above which two texts are deemed to belong in
+                the same cluster. Defaults to None.
+            min_cluster_size (Optional[int], optional): Minimum number of elements in a cluster. Defaults to None.
+
+        Returns:
+            CreateClusterJobResponse: Created clustering job handler
+        """
+
         json_body = {
             "embeddings_url": embeddings_url,
             "threshold": threshold,
             "min_cluster_size": min_cluster_size,
         }
 
         response = self.__request(cohere.CLUSTER_JOBS_URL, json=json_body)
-        return CreateClusterJobResponse(job_id=response['job_id'])
+        return CreateClusterJobResponse(
+            job_id=response['job_id'],
+            poll_fn=self.poll_cluster_job,
+        )
 
     def get_cluster_job(
         self,
         job_id: str,
-    ) -> GetClusterJobResponse:
+    ) -> ClusterJobResult:
+        """Get clustering job results.
+
+        Args:
+            job_id (str): Clustering job id.
+
+        Raises:
+            ValueError: "job_id" is empty
+
+        Returns:
+            GetClusterJobResponse: Clustering job results.
+        """
+
         if not job_id.strip():
             raise ValueError('"job_id" is empty')
 
         response = self.__request(os.path.join(cohere.CLUSTER_JOBS_URL, job_id), method='GET')
-        return GetClusterJobResponse(
+        return ClusterJobResult(
             status=response['status'],
             output_clusters_url=response['output_clusters_url'],
             output_outliers_url=response['output_outliers_url'],
         )
+
+    def poll_cluster_job(
```

Comment:
with this name (and the one below) I would think it just gets the status once. Maybe `wait_for_cluster_job` and `wait_for_result` or something?

Reply:
makes sense, renamed to `wait`